dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueController.java

```diff
@@ -682,6 +686,20 @@ private Period getAndValidatePeriod( String pe )
         return period;
     }
 
+    private void validateOrganisationUnitPeriod( OrganisationUnit organisationUnit, Period period ) throws WebMessageException
+    {
+        Date openingDate = organisationUnit.getOpeningDate();
+
+        Date startDate = period.getStartDate();
+        Date endDate = period.getEndDate();
+        Date closeDate = organisationUnit.getClosedDate() != null ? organisationUnit.getClosedDate() : endDate;
```

Comment:
This logic becomes a bit confusing. I can see it works but it is a bit strange to say closeDate = period end date.

Could we perhaps rather check for null closeDate below in the if statement, since closeDate is nullable?